zebra-state/src/service/non_finalized_state/tests/prop.rs

```diff
@@ -42,16 +44,17 @@ fn finalized_equals_pushed() -> Result<()> {
                                       .ok()
                                       .and_then(|v| v.parse().ok())
                                       .unwrap_or(DEFAULT_PARTIAL_CHAIN_PROPTEST_CASES)),
-    |((chain, end_count, _network) in PreparedChain::default())| {
+    |((chain, end_count, network) in PreparedChain::default())| {
+        let dummy_tree = HistoryTree::new_from_block(network, chain[0].block.clone(), &sapling::tree::Root::default(), None).unwrap();
         let finalized_count = chain.len() - end_count;
-        let mut full_chain = Chain::default();
-        let mut partial_chain = Chain::default();
+        let mut full_chain = Chain::new(dummy_tree.clone());
+        let mut partial_chain = Chain::new(dummy_tree);
 
         for block in chain.iter().skip(finalized_count) {
-            partial_chain.push(block.clone());
+            partial_chain.push(block.clone())?;
         }
         for block in chain.iter() {
-            full_chain.push(block.clone());
+            full_chain.push(block.clone())?;
         }
 
         for _ in 0..finalized_count {
```

Comment:
Testing: Let's add some steps to this test:
1. Make sure `full_chain` and `partial_chain` tip hashes and heights are the same
  - since `partial_chain` skipped some blocks, the rest of the state might not be identical - including the chain history

It's probably not worth comparing any other fields here.